tests/integration/test_application.py

```diff
@@ -254,7 +254,7 @@ def test_non_lifecycle_command_does_not_require_project(monkeypatch, app):
     """Run a command without having a project instance shall not fail."""
     monkeypatch.setattr("sys.argv", ["testcraft", "nothing"])
 
-    class NothingCommand(craft_cli.BaseCommand):
+    class NothingCommand(craft_application.commands.AppCommand):
```

Comment:
This was the issue it caught. Before entering debug mode, `app.run()` was returning an error value. Now in debug mode, it raises an exception. Switching this to an AppCommand fixes it.